src/include/util/hb_arena.h

```diff
@@ -4,15 +4,23 @@
 #include <stdbool.h>
 #include <stddef.h>
 
-typedef struct HB_ARENA_ALLOCATOR_STRUCT {
-  char* memory;
-  size_t capacity;
-  size_t position;
+typedef struct HB_ARENA_PAGE_STRUCT {
+  struct HB_ARENA_PAGE_STRUCT* next;
+  char* start;
+  char* current;
+  char* end;
```

Comment:
Question: Why use pointers here? 

IMO: Using capacity and position is usually a bit clearer and reduces the need for pointer shenanigans.